### PR TITLE
Update used ids on reading an existing model.

### LINKED ISSUE #1806: model.basin.add() in model with existing nodes will cause AssertionError next-time model.node_table() is is accessed
While not to explicitly check available node_id in model with existing nodes, I run this code:

```
        geometry = basin_to_point(basin_polygon=basin_polygon, tolerance=10)
        node = Node(
            meta_categorie="bergend",
            geometry=geometry,
        )
        data = [basin.Profile(level=[0, 1], area=[5, 100]), basin.Area(geometry=[basin_polygon])]
        basin_node = model.basin.add(node=node, tables=data)
```

Inspecting the `model.basin.node.df` I notice node_id =1 has been assigned. Does not exist in the basin-table, but does exist as `FlowBoundary`:
image

Thereby accessing `model.node_table()` will raise an `AssertionError: node_id must be unique`:
image

### PR DESCRIPTION
**Pull request by evetion**
Fixes #1806 and #1804

### CODE REVIEW

File: python/ribasim/ribasim/model.py [hunk 1]
```diff
@@ -224,7 +235,11 @@ def _apply_crs_function(self, function_name: str, crs: str) -> None:
     def node_table(self) -> NodeTable:
         """Compute the full sorted NodeTable from all node types."""
         df_chunks = [node.node.df for node in self._nodes()]
-        df = pd.concat(df_chunks)
+        df = (
+            pd.concat(df_chunks)
+            if df_chunks
+            else pd.DataFrame(index=pd.Index([], name="node_id"))
```
Comment (visr) [inline python/ribasim/ribasim/model.py:241]: This is to handle empty models, without any nodes right? Did you need this for anything? I don't think we test empty models.

I think this fails, just like before, but now with:

```
>>> ribasim.model.NodeTable(pd.Index([], name="node_id"))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: BaseModel.__init__() takes 1 positional argument but 2 were given
>>> ribasim.model.NodeTable(df=pd.Index([], name="node_id"))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "D:\repo\ribasim\Ribasim\.pixi\envs\default\Lib\site-packages\pydantic\main.py", line 193, in __init__
    self.__pydantic_validator__.validate_python(data, self_instance=self)
pydantic_core._pydantic_core.ValidationError: 1 validation error for NodeTable
df
  Value error, Expected <class 'pandas.core.series.Series'> to have name 'node_id', found 'None' [type=value_error, input_value=Index([], dtype='object', name='node_id'), input_type=Index]
    For further information visit https://errors.pydantic.dev/2.8/v/value_error
```

Wouldn't we need all columns with the right dtypes? With this we get an object dtype.

So perhaps best to leave out this change unless needed.

**Review by visr — APPROVED** [review]

File: python/ribasim/ribasim/model.py (hunk 1 above)
Comment (evetion) [reply python/ribasim/ribasim/model.py:241]: Your examples are not what's happening in the code, you need to pass the dataframe as the df argument to the nodetable.

That said, this might not be needed in this PR, it's for the edge case when you read an empty geopackage. And when you call nodetable on a new model, which previously failed.

Comment (visr) [reply python/ribasim/ribasim/model.py:241]: Oh sorry I pasted two prompts, the first once was missing the `df` arguments, but the second one is not.

Comment (evetion) [reply python/ribasim/ribasim/model.py:241]: Your second one passes an index, not a dataframe 😉

Comment (visr) [reply python/ribasim/ribasim/model.py:241]: Ha my bad. Let's merge!
